jobsdb/migration.go

```diff
@@ -218,33 +270,54 @@ func (jd *HandleT) cleanupStatusTables(ctx context.Context, dsList []dataSetT) e
 
 	return jd.WithTx(func(tx *Tx) error {
 		for _, statusTable := range toCompact {
+			table := statusTable.JobStatusTable
+			// clean up and vacuum if not present in toVacuum
+			_, ok := toVacuum[table]
 			if err := jd.cleanStatusTable(
 				ctx,
 				tx,
-				statusTable.JobStatusTable,
+				table,
+				!ok,
 			); err != nil {
 				return err
 			}
+
+			// vacuum full if present in toVacuum
+			if ok {
+				if _, err := tx.ExecContext(ctx, fmt.Sprintf(`VACUUM FULL %[1]q`, table)); err != nil {
+					return err
+				}
+			}
```

Comment:
Full vacuuming shall be performed on its own loop, following but not coupled with the compaction loop.

Reply:
Yeah, my bad 🤦 
Made the changes